x-pack/plugins/maps/public/layers/layer_wizard_registry.ts

```diff
@@ -34,8 +34,14 @@ export function registerLayerWizard(layerWizard: LayerWizard) {
   registry.push(layerWizard);
 }
 
-export function getLayerWizards(): LayerWizard[] {
-  return registry.filter(layerWizard => {
-    return layerWizard.checkVisibility ? layerWizard.checkVisibility() : true;
+export async function getLayerWizards(): Promise<LayerWizard[]> {
+  const promises = registry.map(async layerWizard => {
+    return {
+      ...layerWizard,
+      isVisible: layerWizard.checkVisibility ? await layerWizard.checkVisibility() : true,
```

Comment:
we control all this wizard-code still, so I would just add `checkVisibility()` to all wizards. it avoids introducing this check.